### PR TITLE
chore: add recursive integration test

### PR DESCRIPTION
**Pull request by kobyhallx**
# Description

## Problem\*

Resolves 

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: compiler/integration-tests/package.json [hunk 1]
```diff
@@ -7,19 +7,20 @@
     "build": "echo Integration Test build step",
     "test": "yarn test:browser",
     "test:browser": "web-test-runner",
-    "test:integration:browser": "web-test-runner test/integration/browser/**/*.test.ts",
-    "test:integration:browser:watch": "web-test-runner test/integration/browser/**/*.test.ts --watch",
+    "test:integration:browser": "web-test-runner test/integration/browser/recursion.test.ts",
+    "test:integration:browser:watch": "web-test-runner test/integration/browser/recursion.test.ts --watch",
```
Comment (TomAFrench) [inline compiler/integration-tests/package.json:11]: Need to revert this change before merging so we still run the old tests.

File: compiler/integration-tests/test/integration/browser/recursion.test.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+} from "@noir-lang/noir_wasm";
+import { decompressSync as gunzip } from "fflate";
+import newABICoder, { abiEncode } from "@noir-lang/noirc_abi";
+import initACVM, {
+  executeCircuit,
+  WitnessMap,
+  compressWitness,
+} from "@noir-lang/acvm_js";
+
+// @ts-ignore
```
Comment (TomAFrench) [inline compiler/integration-tests/test/integration/browser/recursion.test.ts:18]: This shouldn't be necessary afaik.
```suggestion
```

File: compiler/integration-tests/test/integration/browser/recursion.test.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
```
Comment (TomAFrench) [inline compiler/integration-tests/test/integration/browser/recursion.test.ts:1]:
```suggestion
```

Can remove after the above.

File: compiler/integration-tests/web-test-runner.config.mjs [hunk 4]
```diff
@@ -23,8 +35,9 @@ export default {
     esbuildPlugin({
       ts: true,
     }),
+    ...debugPlugins,
   ],
-  files: ["test/integration/browser/**/*.test.ts"],
+  files: ["test/integration/browser/recursion.test.ts"],
```
Comment (TomAFrench) [inline compiler/integration-tests/web-test-runner.config.mjs:40]:
```suggestion
  files: ["test/integration/browser/**/*.test.ts"],
```

File: .github/workflows/test-integration.yml [hunk 5]
```diff
@@ -7,6 +7,9 @@ on:
       - ./compiler/integration-tests/**
   schedule:
     - cron: "0 2 * * *" # Run nightly at 2 AM UTC
+  push:
+    branches:
+      - kh-recursive-e2e
```
Comment (TomAFrench) [inline .github/workflows/test-integration.yml:12]:
```suggestion
```

File: compiler/integration-tests/test/integration/browser/recursion.test.ts [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+await api.commonInitSlabAllocator(CIRCUIT_SIZE);
+// Plus 1 needed!
+const crs = await Crs.new(CIRCUIT_SIZE + 1);
+await api.srsInitSrs(
+  new RawBuffer(crs.getG1Data()),
+  crs.numPoints,
+  new RawBuffer(crs.getG2Data())
+);
+
+const acirComposer = await api.acirNewAcirComposer(CIRCUIT_SIZE);
```
Comment (TomAFrench) [inline compiler/integration-tests/test/integration/browser/recursion.test.ts:57]: We can share this with the other integration test in a followup PR.

File: compiler/integration-tests/package.json (hunk 1 above)
Comment (kobyhallx) [reply compiler/integration-tests/package.json:11]: Resolved in 82d5ba70c5d18ac9f7fd1c1fd0dc4c1f9c978aaa

File: compiler/integration-tests/test/integration/browser/recursion.test.ts (hunk 2 above)
Comment (kobyhallx) [reply compiler/integration-tests/test/integration/browser/recursion.test.ts:18]: Still problematic

File: compiler/integration-tests/test/integration/browser/recursion.test.ts (hunk 3 above)
Comment (kobyhallx) [reply compiler/integration-tests/test/integration/browser/recursion.test.ts:1]: Still Problematic

File: compiler/integration-tests/web-test-runner.config.mjs (hunk 4 above)
Comment (kobyhallx) [reply compiler/integration-tests/web-test-runner.config.mjs:40]: Resolved 82d5ba70c5d18ac9f7fd1c1fd0dc4c1f9c978aaa

File: .github/workflows/test-integration.yml (hunk 5 above)
Comment (kobyhallx) [reply .github/workflows/test-integration.yml]: f41b04f: chore: remove on push hook

**Review by TomAFrench — APPROVED** [review]
